pkg/loop/internal/relayer/pluginprovider/contractreader/contract_reader.go

```diff
@@ -291,12 +332,27 @@ func (c *Server) GetLatestValue(ctx context.Context, request *pb.GetLatestValueR
 		return nil, err
 	}
 
-	retVal, err := getContractEncodedType(request.ReadIdentifier, c.impl, false)
+	confidenceLevel, err := confidenceFromProto(request.Confidence)
 	if err != nil {
 		return nil, err
 	}
 
-	confidenceLevel, err := confidenceFromProto(request.Confidence)
+	if request.AsValueType {
+		var retVal values.Value
+		err = c.impl.GetLatestValue(ctx, request.ReadIdentifier, confidenceLevel, params, &retVal)
+		if err != nil {
+			return nil, err
+		}
+
+		encodedRetVal, err2 := EncodeVersionedBytes(retVal, ValuesEncodingVersion)
+		if err2 != nil {
+			return nil, err2
+		}
+
+		return &pb.GetLatestValueReply{VersionedBytes: encodedRetVal}, nil
+	}
```

Comment:
THis is over complicated and also still needs the contract encoded type below or you can loose precision.  You shouldn't need this anyways.  Line 365 encodes it based on the param's version.  If you set that as values.Value on the client,  then encode versioned bytes should just do the encode for you.

Reply:
I've updated with the simpler implementation here